### PR TITLE
Checklist: rollout to other goal types.

### PR DESCRIPTION
**Pull request by taggon**
The checklist is currently available only for a blog type site. This PR will roll out it to other goal types so our customers have a more unified experience.

## How to test

1. Create a new site with any other type but `store`.
2. See if the checklist shows up to you.
3. For a non-blog site, "Publish your first blog post" or "You uploaded a profile picture" items should be omitted.

### CODE REVIEW

File: client/state/selectors/get-site-design-type.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	if ( ! siteIdOrSlug ) {
+		return null;
+	}
+
+	const site = getSite( state, siteIdOrSlug );
+	if ( ! site ) {
+		return null;
+	}
+
+	return get( site, 'options.design_type', '' );
```
Comment (sirreal) [inline client/state/selectors/get-site-design-type.js]: You can leverage `getSiteOption` like this:

https://github.com/Automattic/wp-calypso/blob/8eb8c95465f063b5b56e061d6b8eb5ffc1077d3c/client/my-sites/stats/checklist-banner/index.jsx#L201

Comment (taggon) [reply client/state/selectors/get-site-design-type.js]: I will remove the file in favor of `getSiteOption`. Thanks. :)

**Comment by taggon** [comment]
Rebased

**Comment by fditrapani** [comment]
I tested this with the other options and the checklist shows for me. The copy could use some work but we could leave that for another PR.

**Comment by taggon** [comment]
Rebased

**Comment by taggon** [comment]
> I'd love to see some kind of consolidation of checking if the design type supports the checklist rather than the manual checks in several different places, but I don't consider that a blocker.

I usually do code in that way too. :) But I leave them for now since we're going to roll out the checklist to all users, which means the type checking will be removed in the near future.

Thanks for your review! 👍

**Comment by sirreal** [comment]
This seems to have introduced the regression #26870 fixed by #26876.

**Comment by sirreal** [comment]
This seems to have introduced the regression #26911 fixed by #26912

**Pull request merged**